lib/tasks/fixity_check.rake

```diff
@@ -2,9 +2,75 @@ namespace :scholars_archive do
   desc "repository fixity check"
   task fixity: :environment do
     Rails.logger.warn "Running Hyrax::RepositoryFixityCheckService"
-    # OVERRIDE: From Hyrax, add async option for Fixity Check
+
+    # SETUP: Variables will be used in the case
+    failed_arr = []
+
+    # MAILER: Disable mailer so Fixity won't send out email while running check
+    ActionMailer::Base.perform_deliveries = false
+
+    # CREATE: Make a start time to know when Fixity starts
+    start_time = Time.now
+
+    # OVERRIDE: From Hyrax, add async option for Fixity check
     ::FileSet.find_each do |file_set|
       Hyrax::FileSetFixityCheckService.new(file_set, async_jobs: false).fixity_check
     end
+
+    # CREATE: Make an end time to know when Fixity finish
+    end_time = Time.now
+
+    # QUERY #1: Query data from the ChecksumAuditLog
+    latest_file = ChecksumAuditLog.where("updated_at >= ?", start_time)
+    file_checked = latest_file.count
+    file_passed = latest_file.where("passed = true").count
+    file_failed = latest_file.where("passed = false").count
+
+    # QUERY #2: Get all the ids that failed via checking with fixity
+    latest_file.each do |c|
```

Comment:
You already have all the failed files here: `latest_file.where("passed = false")`
Then you can use a `Array#map` method to extract & capture the `file_set_id`

```
failed_arr = latest_file.where("passed = false").map(&:file_set_id)
```